addons/stock_landed_costs/models/stock_landed_costs.py

```diff
@@ -0,0 +1,244 @@
+    description = fields.Text(string='Item Description', states={'done': [('readonly', True)]})
+    amount_total = fields.Float(compute='_compute_amount_total', string='Total', digits_compute=dp.get_precision('Account'), store=True,
+                                track_visibility='always')
+    state = fields.Selection([('draft', 'Draft'), ('done', 'Posted'), ('cancel', 'Cancelled')], default='draft', readonly=True, track_visibility='onchange', copy=False)
+    account_move_id = fields.Many2one('account.move', string='Journal Entry', readonly=True, copy=False)
+    account_journal_id = fields.Many2one('account.journal', string='Account Journal', required=True)
+
+    @api.depends('cost_lines', 'cost_lines.price_unit', 'cost_lines.cost_id')
+    def _compute_amount_total(self):
+        for landed_cost_line in self:
+            landed_cost_line.amount_total = sum(landed_cost_line.cost_lines.mapped('price_unit'))
+
+    def _create_accounting_entries(self, line, move, qty_out):
+        cost_product = line.cost_line_id and line.cost_line_id.product_id
+        if not cost_product:
+            return False
+        accounts = line.product_id.product_tmpl_id.get_product_accounts()
+        debit_account_id = accounts.get('stock_valuation', False) and accounts['stock_valuation'].id
+        already_out_account_id = accounts.get('stock_output', False) and accounts['stock_output'].id
+        credit_account_id = line.cost_line_id.account_id.id or cost_product.property_account_expense_id.id or cost_product.categ_id.property_account_expense_categ_id.id
+
+        if not credit_account_id:
+            raise UserError(_('Please configure Stock Expense Account for product: %s.') % (cost_product.name))
+        return self._create_account_move_line(line, move, credit_account_id, debit_account_id, qty_out, already_out_account_id)
+
+    def _create_account_move_line(self, line, move, credit_account_id, debit_account_id, qty_out, already_out_account_id):
+        """
+        Generate the account.move.line values to track the landed cost.
+        Afterwards, for the goods that are already out of stock, we should create the out moves
+        """
+        AccountMoveLine = self.env['account.move.line'].with_context(check_move_validity=False)
+        base_line = {
+            'name': line.name,
+            'move_id': move.id,
+            'product_id': line.product_id.id,
+            'quantity': line.quantity,
+        }
+        debit_line = dict(base_line, account_id=debit_account_id)
+        credit_line = dict(base_line, account_id=credit_account_id)
+        diff = line.additional_landed_cost
+        if diff > 0:
+            debit_line['debit'] = diff
+            credit_line['credit'] = diff
+        else:
+            # negative cost, reverse the entry
+            debit_line['credit'] = -diff
+            credit_line['debit'] = -diff
+        AccountMoveLine.create(debit_line)
+        AccountMoveLine.create(credit_line)
```

Comment:
As per my opinion no need to twice call ***create*** method , do it in one ***create*** method